crates/common/src/compile.rs

```diff
@@ -457,34 +458,50 @@ pub struct ContractInfo {
     pub is_dev_contract: bool,
 }
 
+/// Finds the path of the contract with the given name.
+/// Throws error if multiple or no contracts with the same name are found.
+pub fn find_contract_path(target_name: &str, project: &Project) -> Result<PathBuf> {
+    let graph = Graph::resolve(&project.paths)?;
+    let mut target = None;
+
+    for file in graph.files().keys() {
+        let src = std::fs::read_to_string(file)?;
+        let (parsed, _) = solang_parser::parse(&src, 0)
+            .map_err(|_| eyre::eyre!("Failed to parse {}.", file.display()))?;
+
+        for part in parsed.0 {
+            if let SourceUnitPart::ContractDefinition(contract) = part {
+                if let Some(name) = contract.name {
+                    if name.name == target_name {
+                        if target.is_some() {
+                            eyre::bail!(
+                                "Found multiple matching contracts with the name `{}`",
+                                target_name
+                            );
+                        }
+                        target = Some(file);
```

Comment:
can we break here?

Reply:
you mean do not error if we have multiple matches? feels like a footgun